### PR TITLE
Keyboard support

### PR DESCRIPTION
**Pull request by gmazzamuto**
Hi,
I have added a new switch to easily enable keyboard support. That is, the slider handle can be selected and moved using the keyboard. From the demo page it should be clear how to use it; I have updated the gh-pages as well, I'll send a separate PR for that.

Please note that I have committed changes to `src/nouislider.ts` only. I didn't commit changes to the two files `src/nouislider.js` and `src/nouislider.d.ts` since those are generated by the build process. Btw, shouldn't they be removed from the repo altogether?

Hope this change is useful.

### CODE REVIEW

**Comment by tb** [comment]
Ok, I did implement it on my own and not to break current styling you actually `keyboard=true` is good idea because setting tabindex will require setting of `handle.styles.outline` and we probably user who dont care about keyboard dont want to have to style it by default.

Here is implementation I did:

    ngOnInit(): void {
      //...
  
      if (this.keyboard) {
        this.el.nativeElement.querySelectorAll('.noUi-handle').forEach((handle: HTMLElement) => {
          handle.setAttribute('tabindex', '0');
          handle.addEventListener('click', (e: Event) => handle.focus());
        });
      }
    }

Then in demo template:

    <nouislider [connect]="true" [min]="0" [max]="15" [step]="1" [(ngModel)]="someRange" (change)="onChange($event)" [keyboard]="true" (keyup)="onKey($event)"></nouislider>

And demo code:

    onKey(event: KeyboardEvent) {
      console.log('KeyboardEvent',
        'keyCode', event.keyCode,
        'handle index', (<HTMLElement>event.target).getAttribute('data-handle')
      );
    }

I am not sure what to do with the code you wrote in `switch ( e.which )`. I think it should be controlled by the application, but I see also need for providing some default key handler. I think default function needs just 2 parameters: event (to read the keyboard code and handle index) and slider (to get steps and call this.slider.set - which should update slider and trigger model change). I think this function could be part of the config so that users can overwrite that function (it could be called `onkey`). The default implementation could be defined as separate function in the nouislider.ts file.

Let me know what you think.

PS As to the gh-pages I know its not handy atn will try to fix with some npm task that will publish code from master to gh-pages.

**Comment by tb** [comment]
I have optimalized gh-pages build process. Please remove submodule from .git/config and fetch. Demo folder is not id app and deployment is done via `npm run deploy`

**Comment by gmazzamuto** [comment]
I have made some changes following your suggestions. What do you think?

The default implementation can now be overridden by implementing `keyEventToValue`. This function  takes as arguments the event and the current slider value, and returns the new value to be set. I have used your method to retrieve the handle index from the event object (I think that probably this works only with the latest version of nouislider; perhaps it's worth considering upgrading to nouislider 9.0.0).

This should cover most use cases. If you want to add a bit more flexibility, we could provide a property to let the user choose the multiplier for big steps (page up / page down). For example, the [slider widget](http://qwt.sourceforge.net/class_qwt_abstract_slider.html#pub-methods) from the Qwt library has a property named `pageSteps` for page increments.

Btw I have exposed the slider handles, so that users have 100% control if they need some very special implementation.

File: demo/app.component.html [hunk 1]
```diff
@@ -73,4 +73,89 @@
...
+  &#125;
+&#125;;
+</pre>
+
+<pre>&lt;nouislider [config]=&quot;someKeyboardConfig&quot; [(ngModel)]=&quot;someKeyboard&quot;&gt;&lt;/nouislider&gt;</pre>
+Select a handle and use arrow keys or page up / page down.
+<div class="event-logger">
+    Keboard events (key up, key down):
+    <span id="keyup-label" class="h6 label label-default">keyup</span>
+    <span id="keydown-label" class="h6 label label-default">keydown</span>
```
Comment (gmazzamuto) [inline demo/app.component.html]: I agree, I'll fix that

**Comment by gmazzamuto** [comment]
I came up with some interesting improvements, can you please review them?

the implementation is much cleaner now, and there's no need to pass too many values around :)

**Pull request merged**